index.js

```diff
@@ -51,6 +51,19 @@ function doTransform(checker, k) {
    * @return {import("typescript").VisitResult<Node>}
    */
   const transform = function(n) {
+    if (ts.isFunctionDeclaration(n) && n.type && ts.isTypePredicateNode(n.type)) {
```

Comment:
Doesn't this downlevel `x is Foo` &rarr; `void` too? Probably you should add a test like `declare function isString(x: any): x is string` and make sure it doesn't get down-levelled.

From looking at the asserts code in the checker (createTypePredicateFromTypePredicateNode), I think you just need another condition:

```suggestion
    if (ts.isFunctionDeclaration(n) && n.type && ts.isTypePredicateNode(n.type) && node.assertsModifier) {
```

Reply:
No, it doesn't because that's not a function declaration apparently?

I agree with you however, so I'll chuck a test in there & update the condition (ironically that was the condition that I started with, and worked backward from).